src/libraries/System.IO.Compression/tests/ZipArchive/zip_InvalidParametersAndStrangeFiles.cs

```diff
@@ -128,6 +129,66 @@ public static async Task ZipArchiveEntry_InvalidUpdate(string zipname)
             }
         }
 
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "Fix not shipped for .NET Framework.")]
+        [ActiveIssue("https://github.com/dotnet/runtime/issues/32316", TestRuntimes.Mono)]
+        public static void ZipArchiveEntry_AppContext_Switch_SuppresHeaderValidation(bool suppressValidation)
+        {
+            RemoteExecutor.Invoke((suppressString) =>
+            {
+                bool suppress = bool.Parse(suppressString);
+                MemoryStream stream = populateStream().Result;
+                PatchDataRelativeToFileName(Encoding.ASCII.GetBytes(s_tamperedFileName), stream, 8);  // patch uncompressed size in file header
+                using (ZipArchive archive = new ZipArchive(stream, ZipArchiveMode.Read))
+                {
+                    // Set the AppContext Switch
+                    AppContext.SetSwitch("System.Compression.ZipArchiveEntry.SuppressHeaderValidation", suppress);
+                    ZipArchiveEntry e = archive.GetEntry(s_tamperedFileName);
+                    if (suppress)
+                    {
+                        using (MemoryStream ms = new MemoryStream())
+                        using (Stream source = e.Open())
+                        {
+                            source.CopyTo(ms);
+                            Assert.Equal(ms.Position, ms.Length);  // Just making sure it was readable
+                        }
+                    }
+                    else
+                    {
+                        Assert.Throws<InvalidDataException>(() => e.Open());
+                    }
+                }
+            }, suppressValidation.ToString()).Dispose();
+        }
+
+        [Fact]
+        public static async Task LargeArchive_DataDescriptor_Read_NonZip64_FileLengthGreaterThanIntMax()
+        {
+            MemoryStream stream = await LocalMemoryStream.readAppFileAsync(strange("fileLengthGreaterIntLessUInt.zip"));
```

Comment:
"strange" ?

Reply:
It's a folder in the testdata nuget. This method gets a full path to that folder.

https://github.com/dotnet/runtime/blob/2ef6db0bf90584c4c9788441c562ec606a3438a3/src/libraries/Common/tests/System/IO/Compression/ZipTestHelper.cs#L17